### PR TITLE
Check that a Postmodel exists before attempting to update it

### LINKED ISSUE #6530: Crash report 8.1: NPE in PostStore.updatePost()
```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'void org.wordpress.android.fluxc.model.PostModel.setDateLocallyChanged(java.lang.String)' on a null object reference
       at org.wordpress.android.fluxc.store.PostStore.updatePost(PostStore.java:548)
       at org.wordpress.android.fluxc.store.PostStore.onAction(PostStore.java:365)
       at java.lang.reflect.Method.invoke(Method.java)
       at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:485)
```

The crash is in FluxC, but we should understand why we're ending up with a `null` `PostModel` when sending an `UPDATE_POST` action from WPAndroid.

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #6530 

To test:
1. start a new draft, put lots of images in there (big, 2mb images, at lest 6 or 8 of them) by using the multiple selector in the media picker
2. exit the editor
3. in the posts list, tap on `DELETE/TRASH` button
4. see the snackbar appear, wait until it disappears: if the app doesn't crash, it means this worked :)

cc @aforcier

### CODE REVIEW

**Comment by aforcier** [comment]
We do similar handling [further down in `OnMediaUploaded`](https://github.com/wordpress-mobile/WordPress-Android/blob/db9147201b9d45cef0792f47063ddeb641706dac/WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java#L706-L713), shall we make the same change there too?

**Comment by mzorz** [comment]
good catch, addressed in 3a2b7bb

File: WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java [hunk 1]
```diff
@@ -700,17 +705,22 @@ public void onMediaUploaded(OnMediaUploaded event) {
...
+                                    updatedPost = updatePostWithCurrentlyFailedUploads(updatedPost);
+                                    // finally, save the PostModel
+                                    if (updatedPost != null) {
+                                        mDispatcher.dispatch(PostActionBuilder.newUpdatePostAction(updatedPost));
+                                        mPostUploadHandler.upload(updatedPost);
+                                    }
+                                }
+
 
                                 // TODO Should do some extra validation here
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java]: This comment actually belonged to the `mPostHandler.upload()` part, so it should be moved above along with that part of the code.

Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java]: agh - sorry, done in 4a21851

**Comment by aforcier** [comment]
Looks good! I was able to reproduce inconsistently, can no longer reproduce at all after many attempts with this branch.

:shipit:

**Pull request merged**